code/data_structures/src/queue/priority_queue/priority_queue.cpp

```diff
@@ -0,0 +1,107 @@
+// C++ implementation of Priority Queue
+
+#include <iostream>
+#include <cstdio>
+#include <cstring>
+#include <cstdlib>
+ 
+// Node Declaration
+struct node {
+	int priority;
+	int info;
+	node *link;
+};
+
+class PriorityQueue {
+    private:
+        node *front;
+    public:
+        PriorityQueue() {
+            front = nullptr;
+        }
+        // Insert into Priority Queue
+        void add(int item, int priority) {
+            node *tmp, *q;
+            tmp = new node;
+            tmp->info = item;
+            tmp->priority = priority;
+            if (front == nullptr || priority < front->priority){
+                tmp -> link = front;
+                front = tmp;
+            }
+            else {
+                q = front;
+                while (q->link != nullptr && q->link -> priority <= priority)
+                    q = q->link;
+
+                tmp->link = q->link;
+                q->link = tmp;
+            }
+        }
+        
+        // Delete from Priority Queue
+        void remove() {
+            node *tmp;
+            if(front == nullptr)
+                std::cout << "Queue Underflow\n";
```

Comment:
Don't print anything out; raise an exception or return a boolean instead; handle errors in `main`.

Reply:
done